hpat/distributed_analysis.py

```diff
@@ -306,6 +306,17 @@ def _analyze_call(self, lhs, func_var, args, array_dists):
             self._analyze_call_np_concatenate(lhs, args, array_dists)
             return
 
+        if call_list == ['permutation', 'random', np]:
```

Comment:
This piece of code shouldn't be part of this patch because shuffling of arrays with permutation is not implemented here.